WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductImageThumbnail.swift

```diff
@@ -7,27 +7,39 @@ struct ProductImageThumbnail: View {
     private let scale: CGFloat
     private let productImageCornerRadius: CGFloat
     private let foregroundColor: Color
+    private let cache: ImageCache = ImageCache.default
+    private let usesDefaultCache: Bool
 
     /// Image processor to resize images in a background thread to avoid blocking the UI
     ///
     private var imageProcessor: ImageProcessor {
-        ResizingImageProcessor(referenceSize:
-                .init(width: productImageSize * scale,
-                      height: productImageSize * scale),
-                               mode: .aspectFill
-        )
+        if usesDefaultCache {
+            DefaultImageProcessor()
+        } else {
+            ResizingImageProcessor(referenceSize:
+                    .init(width: productImageSize * scale,
+                          height: productImageSize * scale),
+                                   mode: .aspectFill)
+        }
     }
 
-    init(productImageURL: URL?, productImageSize: CGFloat, scale: CGFloat, productImageCornerRadius: CGFloat = 0, foregroundColor: Color) {
+    init(productImageURL: URL?,
+         productImageSize: CGFloat,
+         scale: CGFloat,
+         productImageCornerRadius: CGFloat = 0,
+         foregroundColor: Color,
+         usesDefaultCache: Bool = false) {
         self.productImageURL = productImageURL
         self.productImageSize = productImageSize
         self.scale = scale
         self.productImageCornerRadius = productImageCornerRadius
         self.foregroundColor = foregroundColor
+        self.usesDefaultCache = usesDefaultCache
     }
 
     var body: some View {
-        KFImage.url(productImageURL)
+        KFImage
+            .url(productImageURL)
```

Comment:
```suggestion
            .url(productImageURL)
            .cacheOriginalImage()
```

What would you think of using `cacheOriginalImage`, would it achieve the same result?

We could create a conditional flag from `ProductImageThumbnail` which we could pass from itemRow and itemCard, which would conditionally set `cacheOriginalImage()` modifier:

> If set and an `ImageProcessor` is used, Kingfisher will try to cache both the final result and original
    /// image. Kingfisher will have a chance to use the original image when another processor is applied to the same
    /// resource, instead of downloading it again

It should ensure that not only resized image we use for product list is cached but also original image, so it could be used for resized version in the cart list.

Reply:
Thanks for the recommendation, that's much cleaner, and reduces the PR considerably 💯 

I made the changes on https://github.com/woocommerce/woocommerce-ios/pull/13921/commits/e60bf6e7b774ae7b992d7e00486c7bca0b184f2a , then reverted the logic needed for the initial implementation in the following commits, with this, the `ProductImageThumbnail` continues using cache for the POS views, but keeps the existing behaviour for other uses (using it everywhere looks like a good quick project for hack week).

I wasn't sure of how the combination of `.url()` and `.cacheOriginalImage()` would work together, but it's covered:
> When you set the same URL with KFImage, it will be loaded from the cache [...] When converting to source, a LocalFileImageDataProvider returns the image if the URL points to a local file, or uses the network to download it otherwise.

In order to test, the easier way is to [add an .onSuccess() modifier](https://github.com/woocommerce/woocommerce-ios/blob/bdb7a44f4283ae97f20a9aaaa589862c9dda4ee6/WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductImageThumbnail.swift#L43) to the `KFImage` so the system tells us where the image comes from:

``` diff
.resizable()
+.onSuccess({ result in
+  switch result.cacheType {
+    default:
+      debugPrint("🍍 Image \(productImageURL?.absoluteString) loaded from \(result.cacheType)")
+    }
+})
```

And we get the Image origin both when loaded on Item List and when tapping the items to add to cart. We can also note it uses the original or scaled versions as needed.
```
"🍍 Image Optional(\"https://indiemelon.mystagingwebsite.com/wp-content/uploads/2024/01/IMG_0316.png\") loaded from disk"
"🍍 Image Optional(\"https://indiemelon.mystagingwebsite.com/wp-content/uploads/2023/10/img_0005-scaled.jpeg\") loaded from disk"
"🍍 Image Optional(\"https://indiemelon.mystagingwebsite.com/wp-content/uploads/2023/12/img_0250-scaled.jpeg\") loaded from disk"
```

Reply:
Thanks for this explanation and ways of confirming cache behavior!